src/core/geometry/qgsgeometry.h

```diff
@@ -2646,6 +2646,48 @@ class CORE_EXPORT QgsGeometry
      */
     QgsGeometry makeValid( Qgis::MakeValidMethod method = Qgis::MakeValidMethod::Linework, bool keepCollapsed = false ) const SIP_THROW( QgsNotSupportedException );
 
+    /**
+     * Returns the orientation of the polygon.
+     * \warning Only the first exterior ring is taken to perform this operation. In case of degenerate orders,
+     * you have to perform in deep verification.
+     *
+     * \warning returns Qgis::AngularDirection::Clockwise if the geometry is not a polygon type
+     *
+     * \since QGIS 3.36
+     */
+    Qgis::AngularDirection polygonOrientation() const;
```

Comment:
those new methods will deserve to receive dedicated unit tests (beyond indirectly testing them in the context of maptools)

Reply:
done in Python tests